### PR TITLE
output-level-tiverma

### PR DESCRIPTION
**Pull request by tiverma-msft**
The quiet mode.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -185,6 +192,7 @@ func init() {
 
 	rootCmd.PersistentFlags().Float64Var(&cmdLineCapMegaBitsPerSecond, "cap-mbps", 0, "Caps the transfer rate, in megabits per second. Moment-by-moment throughput might vary slightly from the cap. If this option is set to zero, or it is omitted, the throughput isn't capped.")
 	rootCmd.PersistentFlags().StringVar(&outputFormatRaw, "output-type", "text", "Format of the command's output. The choices include: text, json. The default value is 'text'.")
+	rootCmd.PersistentFlags().StringVar(&outputVerbosityRaw, "output-level", "default", "Define the output verbosity, available levels: ESSENTIAL(Job ID, Log File Location, Job Summary), and QUIET(no output).")
```
Comment (nakulkar-msft) [inline cmd/root.go:195]: Output levels should be in regular case. Check the values for output type.

File: common/lifecyleMgr.go [hunk 2]
```diff
@@ -276,6 +278,12 @@ func (lcm *lifecycleMgr) Info(msg string) {
 }
 
 func (lcm *lifecycleMgr) Prompt(message string, details PromptDetails) ResponseOption {
+
+	if shouldQuietMessage(outputMessage{msgType: eOutputMessageType.Prompt()}, lcm.OutputVerbosityType) {
```
Comment (nakulkar-msft) [inline common/lifecyleMgr.go:282]: Why is this required after you've made sure conflicting options are not allowed?

File: common/lifecyleMgr.go [hunk 3]
```diff
@@ -400,15 +408,19 @@ func (lcm *lifecycleMgr) processOutputMessage() {
...
 		msgToPrint := <-lcm.msgQueue
 
-		switch lcm.outputFormat {
-		case EOutputFormat.Json():
-			lcm.processJSONOutput(msgToPrint)
-		case EOutputFormat.Text():
-			lcm.processTextOutput(msgToPrint)
-		case EOutputFormat.None():
+		if shouldQuietMessage(msgToPrint, lcm.OutputVerbosityType) {
 			lcm.processNoneOutput(msgToPrint)
```
Comment (nakulkar-msft) [inline common/lifecyleMgr.go:412]: You should discard the message here and continue rather than passing message to processNoneOutput

**Review by nakulkar-msft — CHANGES_REQUESTED** [review]

File: cmd/root.go (hunk 1 above)
Comment (tiverma-msft) [reply cmd/root.go:195]: okay, changed

File: common/lifecyleMgr.go (hunk 2 above)
Comment (tiverma-msft) [reply common/lifecyleMgr.go:282]: Conflicting options are not allowed = the user can't pass in quiet or essential mode with overwrite=prompt.
What I'm doing here is checking if the current output-level is quiet or essential. If so, I'm passing yes as the answer to all prompts. This will not be the case when output-level=default.
This is logically the same as "if output-level is not 'default', return yes as the answer to the prompt"

File: common/lifecyleMgr.go (hunk 3 above)
Comment (tiverma-msft) [reply common/lifecyleMgr.go:412]: No actually that would lead to our code never ending. When the code ends, it sends a message with eOutputMessageType.Error() or eOutputMessageType.EndOfJob(), and that's how we know when to call 
`lcm.closeFunc();
os.Exit(int(msgToOutput.exitCode))`

If we just discard the message, that would never be triggered, and our code would hang

Comment (tiverma-msft) [reply common/lifecyleMgr.go:412]: For all other eOutputMessageType (s) I could very well have discarded the message, but I have kept it for the times when it is exit or error

Comment (tiverma-msft) [reply common/lifecyleMgr.go:412]: `func (lcm *lifecycleMgr) processNoneOutput(msgToOutput outputMessage)`
This function will automatically discard the rest of the types

File: cmd/copy.go [hunk 4]
```diff
@@ -853,6 +853,13 @@ func (raw rawCopyCmdArgs) cook() (CookedCopyCmdArgs, error) {
 
 	cooked.dryrunMode = raw.dryrun
 
+	if cooked.ForceWrite == common.EOverwriteOption.Prompt() && (azcopyOutputVerbosity == common.EOutputVerbosity.Quiet() || azcopyOutputVerbosity == common.EOutputVerbosity.Essential()) { //TODO add to other commands
```
Comment (tiverma-msft) [inline cmd/copy.go:856]: don't allow with dry run either

File: common/lifecyleMgr.go (hunk 2 above)
Comment (tiverma-msft) [reply common/lifecyleMgr.go:282]: closing this now

File: common/lifecyleMgr.go (hunk 3 above)
Comment (tiverma-msft) [reply common/lifecyleMgr.go:412]: closing this

File: cmd/copy.go (hunk 4 above)
Comment (tiverma-msft) [reply cmd/copy.go]: done #

**Review by siminsavani-msft — APPROVED** [review]
I would also suggest writing tests to check the output. Take a look at how dry-run is tested to get a foundation for testing output level. Otherwise, LGTM!

File: common/lifecyleMgr.go (hunk 2 above)
Comment (nakulkar-msft) [reply common/lifecyleMgr.go:282]: Unresolving. This is unreachable code.

**Comment by tiverma-msft** [comment]
https://github.com/Azure/azure-storage-azcopy/pull/1754/files/0c3337f7104826194dae19c0dd3cba7267b336cf#r864458485
Previous PR

**Review by nakulkar-msft — APPROVED** [review]

File: common/lifecyleMgr.go (hunk 2 above)
Comment (tiverma-msft) [reply common/lifecyleMgr.go]: makes sense, changing this.

**Review by mohsha-msft — APPROVED** [review]
